src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ErrorReportController.kt

```diff
@@ -1,20 +1,55 @@
 package org.imperial.mrc.hint.controllers
 
+import org.imperial.mrc.hint.AppProperties
 import org.imperial.mrc.hint.clients.FuelFlowClient
+import org.imperial.mrc.hint.logging.GenericLoggerImpl
 import org.imperial.mrc.hint.models.ErrorReport
-import org.springframework.http.*
-import org.springframework.web.bind.annotation.PostMapping
-import org.springframework.web.bind.annotation.RequestBody
-import org.springframework.web.bind.annotation.ResponseBody
-import org.springframework.web.bind.annotation.RestController
+import org.imperial.mrc.hint.service.ProjectVersionService
+import org.slf4j.LoggerFactory
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.util.*
 
 @RestController
-class ErrorReportController(private val fuelFlowClient: FuelFlowClient)
+class ErrorReportController(
+    private val fuelFlowClient: FuelFlowClient,
+    private val projectVersionService: ProjectVersionService,
+    private val appProperties: AppProperties,
+)
 {
+    private val logger = GenericLoggerImpl(LoggerFactory.getLogger(ErrorReportController::class.java))
+
     @PostMapping("/error-report")
     @ResponseBody
-    fun postErrorReport(@RequestBody errorReport: ErrorReport): ResponseEntity<String>
+    fun postErrorReport(@RequestBody errorReport: ErrorReport,
```

Comment:
Could just include projectId in the `ErrorReport` model I guess, rather than including in the qs. I'm a bit surprised it isn't in there already!

Reply:
Thanks, yeah not sure why I didn't put it there in the first place. I've updated this too